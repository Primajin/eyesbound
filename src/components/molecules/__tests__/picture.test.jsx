import React from 'react';
import {act, create} from 'react-test-renderer';

import Picture from '../picture.jsx';

describe('Picture', () => {
	it('renders correctly without extra props', () => {
		const data = {title: 'title', image: {}};
		const component = create(<Picture data={data}/>);
		expect(component).toMatchSnapshot();
	});

	it('renders correctly with preferThumbnails true', () => {
		let data = {title: 'title', image: {gatsbyImageData: {}}};
		const layout = 'FIXED';
		const preferThumbnails = true;
		const size = {height: 123, width: 456};
		const component = create(<Picture data={data} layout={layout} preferThumbnails={preferThumbnails} size={size}/>);
		expect(component).toMatchSnapshot();
		data = {title: 'title', image: {thumbnails: {thumbnail: {gatsbyImageData: {}}}}};
		act(() => {
			component.update(<Picture data={data} layout={layout} preferThumbnails={preferThumbnails} size={size}/>);
		});
	});

	it('renders correctly with preferThumbnails false', () => {
		let data = {title: 'title', image: {gatsbyImageData: {}}};
		const layout = 'FIXED';
		const preferThumbnails = false;
		const size = {height: 123, width: 456};
		const component = create(<Picture data={data} layout={layout} preferThumbnails={preferThumbnails} size={size}/>);
		expect(component).toMatchSnapshot();
		data = {title: 'title', image: {thumbnails: {thumbnail: {gatsbyImageData: {}}}}};
		act(() => {
			component.update(<Picture data={data} layout={layout} preferThumbnails={preferThumbnails} size={size}/>);
		});
	});
});
