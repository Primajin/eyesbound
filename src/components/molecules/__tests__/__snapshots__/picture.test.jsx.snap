// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Picture renders correctly with preferThumbnails false 1`] = `
<picture>
  <img
    alt="title"
    css={
      Object {
        "map": undefined,
        "name": "m2yydh",
        "next": undefined,
        "styles": "
	padding-top: 66.73913043478261%;
",
        "toString": [Function],
      }
    }
    height={123}
    image={
      Object {
        "layout": "FIXED",
      }
    }
    width={456}
  />
</picture>
`;

exports[`Picture renders correctly with preferThumbnails true 1`] = `
<picture>
  <img
    alt="title"
    css={
      Object {
        "map": undefined,
        "name": "m2yydh",
        "next": undefined,
        "styles": "
	padding-top: 66.73913043478261%;
",
        "toString": [Function],
      }
    }
    height={123}
    image={
      Object {
        "layout": "FIXED",
      }
    }
    width={456}
  />
</picture>
`;

exports[`Picture renders correctly without extra props 1`] = `
<picture>
  <img
    alt="title"
    css={
      Object {
        "map": undefined,
        "name": "m2yydh",
        "next": undefined,
        "styles": "
	padding-top: 66.73913043478261%;
",
        "toString": [Function],
      }
    }
    image={
      Object {
        "layout": "CONSTRAINED",
      }
    }
  />
</picture>
`;
